### PR TITLE
ina238: set shunt calibration to desired value if readback is incorrect

### PR DESCRIPTION
**Pull request by RomanBapst**
### Solved Problem
In several log files (about 3 to this point) we have seen what looks like the shunt calibration being lost by the current sensor.
The issue was apparent by the fact that the current reading did not make any sense for the specific vehicle and also we verified by doing the maths, that the reported current matches the case when the shunt calibration was reset to default value.

It's not quite clear yet what is actually causing the reset, investigations are still ongoing.

image-20231016-090108

### Changelog Entry
For release notes:
```
INA238: re-apply sensor shunt calibration if readback is not correct
```

### CODE REVIEW

**Comment by dagar** [comment]
The general pattern I've tried to adopt for most of the other I2C and SPI drivers is to collect all the configuration we care about in a table. Then periodically when operating the config is verified. If any of the registers are misconfigured, or we haven't had a successful read in a bit then a full reset is issued.

So basically, trust nothing and as much as possible be in a position to fully self recover. This is especially important for the external I2C devices.

https://github.com/PX4/PX4-Autopilot/blob/6eae9fb37107ba55d52ab7e3189b1c403cea45a6/src/drivers/magnetometer/isentek/ist8310/IST8310.hpp#L106-L112

For the configuration table I generally try to capture everything we need to set (or clear) from a reset state, but otherwise consider any other bits none of our concern.

**Comment by RomanBapst** [comment]
@dagar Ok, I can restructure it to reflect the general pattern.

**Comment by dagar** [comment]
> @dagar Ok, I can restructure it to reflect the general pattern.

Sounds good, let me know if you'd like any help. I also think we should drop the unnecessary parameters like INA228_CONFIG while we're at it.

Do you want to merge this PR in the meantime?

**Comment by RomanBapst** [comment]
@dagar 

> Do you want to merge this PR in the meantime?
No, I will push to this.

**Comment by RomanBapst** [comment]
@dagar Could you take a look again please?

File: src/drivers/power_monitor/ina238/ina238.h [hunk 1]
```diff
@@ -54,7 +56,7 @@ using namespace time_literals;
 #define INA238_INIT_RETRY_INTERVAL_US       500000
 
 /* INA238 Registers addresses */
-#define INA238_REG_CONFIG                   (0x00)
+//#define INA238_REG_CONFIG                   (0x00)
 #define INA238_REG_ADCCONFIG                (0x01)
```
Comment (dagar) [inline src/drivers/power_monitor/ina238/ina238.h:60]: still duplicated in ina238_registers.hpp?

**Comment by RomanBapst** [comment]
@dagar I addressed you comments.

**Review by dagar — APPROVED** [review]

**Pull request merged**